spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/ComposedTaskRunnerConfiguration.java

```diff
@@ -102,17 +107,28 @@ private static void addTaskExplorer(
 			String taskName
 	) {
 		logger.debug("addTaskExplorer:{}", taskName);
-		String propertyName = String.format("app.%s.spring.cloud.task.tablePrefix", taskName);
-		String prefix = properties.getComposedTaskAppProperties().get(propertyName);
-		if (prefix == null) {
-			prefix = env.getProperty(propertyName);
-		}
+		List<String> propertyNames = new ArrayList<>();
+		RelaxedNames relaxedNames = RelaxedNames.forCamelCase("tablePrefix");
```

Comment:
I would do:
```
RelaxedNames relaxedNames = RelaxedNames.forCamelCase(String.format("app.%s.spring.cloud.task.tablePrefix", taskName));
relaxedNames.forEach(propertyNames::add);
```
This way you are sure in the case of a strange taskName you still match a valid property.
Keep in mind that env.getProperty uses relaxed matching.

Reply:
Unfortunately that will not work.   The `taskName` portion of the property id is also adjusted along with the `tablePrefix` if you pass the entire id into the forCamelCase.  Thus causing a miss when looking up the property.

Reply:
The taskName doesn't change during that method?

Reply:
Doesn;t names like foo-a and fooA clash in other areas?

Reply:
Only if we process the whole property ID via the RelaxedNames will the foo-a get updated.